src/charm.py

```diff
@@ -196,8 +196,11 @@ def _prometheus_configurations(self):
             configured_jobs.append(job)
 
         alerts = list(self._metrics_providers.alerts().values())
+        alert_groups = {"groups": []}  # type: ignore
+        for entry in alerts:
+            alert_groups["groups"] = alert_groups["groups"] + entry["groups"]
 
-        return {"scrape_jobs": configured_jobs, "alert_rules": alerts[0] if alerts else []}
+        return {"scrape_jobs": configured_jobs, "alert_rules": alert_groups if alerts else []}
```

Comment:
This looks subtly wrong, but I'm waiting for it to deploy to test.

`"alert_rules": alert_groups if alert_groups["groups"] else []}` 

Or:

```python
alert_rules = {"groups: []"} if alerts else {}
...

..., "alert_rules": alert_rules or []}
```

?

The variable naming is a little confusing since it's used as a value in a dict with a different name immediately after, but mostly it's confusing because we return a `dict` if `alerts` isn't empty, and a `list` if it is. It should be the same type returned no matter what.

Reply:
You are right. I think there is an issue here. Let me see what a charm with no alert rules is supposed to return.

Reply:
Fixed and pushed.